iml-gui/crate/src/components/breadcrumbs.rs

```diff
@@ -0,0 +1,103 @@
+use crate::{Msg, Page};
+
+use seed::{
+    a, attrs, div,
+    dom_types::{At, El, Node, UpdateEl, View},
+    li, ol,
+    prelude::IndexMap,
+};
+
+use std::cmp::PartialEq;
+use std::collections::LinkedList;
+
+pub struct Model<Crumb> {
```

Comment:
I'm not convinced that we need a separate model for this and it couldn't live in the top-level model in lib. The messages are already on the top-level as well.

Reply:
I think we would need  a wrapping type (struct) anyway. We could call it something different than Model.
The Msg is only needed for the view, but  AFAIK it needs to be defined next to the "model", as well as other methods (goto, goback). Need to think how not to put everything in one top-level module and at the same time not make it overly complicated :)

Reply:
Maybe use a trait instead of a struct.

Reply:
What we ultimately have in the model is a single top-level state atom. I suppose it's really subjective whether we use a separate discrete struct as a model or not, since in the end it serves the same purpose. As long as we keep the messages in the top level, it should work ok.